packages/next/src/server/app-render/app-render.tsx

```diff
@@ -2537,26 +2522,27 @@ async function prerenderToStream(
         )
 
         function finalRenderOnError(err: unknown) {
-          if (err === abortReason) {
-            reactServerIsDynamic = true
-            return PRERENDER_COMPLETE
-          } else if (isPrerenderInterruptedError(err)) {
+          if (isPrerenderInterruptedError(err)) {
             reactServerIsDynamic = true
             return err.digest
+          } else if (finalRenderFlightSignal.aborted) {
+            reactServerIsDynamic = true
+            return PRERENDER_COMPLETE
           }
 
           return serverComponentsErrorHandler(err)
         }
 
         function SSROnError(err: unknown, errorInfo?: ErrorInfo) {
           if (
-            isAbortReason(err, abortReason) ||
-            isPrerenderInterruptedError(err)
+            isPrerenderInterruptedError(err) ||
```

Comment:
I thought so at first but there is the case where during RSC you interrupt and these errors get serialized and encountered by SSR but SSR hasn’t aborted yet. We want to track these for allowed dynamic detection and need to do this before we have aborted.

this may not strictly be needed in the PPR case because these errors don’t get serialized when you pretender but since that’s an impl detail i figured I’d leave it in since semantically it is correct that if we did send this kind of error from RSC to SSR we would want to track it as dynamic